.github/workflows/build-container-image.yml

```diff
@@ -0,0 +1,61 @@
+name: build-container-image
+
+on:
+  push:
+    branches:
+      - main
+  release:
+    types:
+      - created
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    env:
+      CONTAINER_REGISTRY_USERNAME: ${GITHUB_ACTOR}
+      CONTAINER_REGISTRY_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Cache Maven Packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('./**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: Build necessary modules
+        run: mvn install -pl context-monitoring -am
+
+      - name: Get image tag
+        if: ${{ github.event_name == 'push' }}
+        id: get_image_tag
+        run: echo "::set-output name=sha8::$(echo ${GITHUB_SHA} | cut -c1-8)"
+
+      - name: Build Image
+        if: ${{ github.event_name == 'push' }}
+        env:
+          IMAGE_TAG: ${{ steps.get_image_tag.outputs.sha8}}
+        run: mvn compile jib:build -pl smartclide-monitoring -Djib.to.auth.username="${CONTAINER_REGISTRY_USERNAME}" -Djib.to.auth.password="${CONTAINER_REGISTRY_TOKEN}" -Djib.to.tags="${IMAGE_TAG}"
+
+      - name: Get latest release info
+        if: ${{ github.event_name == 'release' }}
+        id: latest_release_info
+        uses: jossef/action-latest-release-info@v1.1.0
```

Comment:
Could you also just extract the tag / release name from the gitref, like with the commit sha?

`run: echo ::set-output name=TAG_NAME::${GITHUB_REF#refs/tags/}`

That would avoid having to import an untrusted third party action

Reply:
actually this works:
`run: echo "::set-output name=RELEASE_NAME::${{ github.event.release.name }}"`